### ISSUE TITLE
Possible to bind v to another editor

### ISSUE THREAD

**Issue by vvatikiotis** [gwsw/less#571]
Read the man page and saw that LESSEDIT default is  %E ?lm+%lm. %g, where %E is the editor? Is it possible to substitute %E with another editor, without changing or relying on $EDITOR or $VISUAL?

Something like LESSEDIT="code ?lm+%lm. %g", where `code` is VSCode. I've tried it, no luck.

**Comment by gwsw** [comment]
Yes, something like that should work. I'm not very familiar with VSCode, but I don't think it accepts the ```+linenumber``` syntax, does it? You probably need to replace that with whatever the syntax is for VSCode.

**Comment by vvatikiotis** [comment]
I'm trying to set it to a bash script though, like the one used by `lessfilter`,  something like the following:

`LESSEDIT="openfilter %g"`

and the executable `openfilter` script is:

```bash
mime=$(file -Lbs --mime-type "$1")
category=${mime%%/*}
kind=${mime##*/}
ext=${1##*.}

if [ "$ext" = ipynb ]; then
  code -n "$1"
elif [ "$kind" = json ]; then
  code -n "$1"
elif [ "$kind" = csv ]; then
  code -n "$1"
elif [ "$ext" = md ]; then
  nvim "$1"
else
  nvim "$1"
fi
```

Viewing, say, a json file, works via .lessfilter, and views the file via the specified json viewer (specified in the `.lessfilter` script). Pressing `v` though (in `less` viewer mode) doesn't take into account the `.openfilter` script and it seems to revert to the json viewer specified in `.lessfilter`. 


Strange is thing is if I set `LESSEDIT="code %g"` and press `v` in `less` viewer mode it opens the said json file in the json terminal viewer (specified in `.lessfilter`) **and** in VScode (specified in the above script).

Any ideas? Thanks!

**Comment by gwsw** [comment]
Sorry, I don't know what lessfilter is. Sounds like something related to your LESSOPEN script? Do things work differently if you invoke less with the ```-L``` option?

**Comment by vvatikiotis** [comment]
It's exactly that. `.lessfilter` is part of the `lesspipe.sh`  (https://github.com/wofr06/lesspipe) solution to instructing `less` to use other terminal programs for viewing files. Just set `LESSOPEN` to a pipe to the `lesspipe.sh` script, which is a input preprocessor. Nothing fancy, all clearly documented.

Invoking `less` with `-L disables this functionality.

Regardless, I'm referring to `.lessfilter` and `lesspipe.sh` as an example or reference in my previous post https://github.com/gwsw/less/issues/571#issuecomment-2352321976 , but what I managed to do is turn the focus to something unrelated to my original issue. Which really is, can I use an executable shell script like the one above, when setting `LESSEDIT`? Can I `export LESSEDIT='my-script.sh %g`?

**Comment by gwsw** [comment]
>Can I ```export LESSEDIT='my-script.sh %g```

That should work, except that since "```.```" is a meta-character in prompt strings, you need to escape it with a backslash: 
```export LESSEDIT='my-script\.sh %g```.

**Comment by vvatikiotis** [comment]
The sun is shining :D Thank you!

**Comment by vvatikiotis** [comment]
It works as you said. Thanks. 

I'm closing the issue (hope this is proper github etiquette).
